src/Commands/core/DrupalDependenciesCommands.php

```diff
@@ -0,0 +1,307 @@
+                '@config' => $config,
+            ]));
+            return null;
+        }
+
+        $this->buildTree($config);
+
+        if (empty($options['format'])) {
+            return $this->drawTree($config);
+        }
+
+        return new UnstructuredData($this->tree);
+    }
+
+    #[CLI\Hook(type: HookManager::ARGUMENT_VALIDATOR, target: 'why:config')]
+    public function validateDependentsOfConfig(CommandData $commandData): void
+    {
+        $configName = $commandData->input()->getArgument('config');
+        $configManager = \Drupal::getContainer()->get('config.manager');
+        if (!$configManager->loadConfigEntityByName($configName)) {
+            throw new \InvalidArgumentException(dt('Invalid @config config entity', [
+                '@config' => $configName,
+            ]));
+        }
+    }
+
+    /**
+     * @param string $dependency
+     * @param array $path
+     */
+    protected function buildTree(string $dependency, array $path = []): void
+    {
+        $path[] = $dependency;
+        foreach ($this->dependents[$dependency] as $dependent) {
+            if (isset($this->relation[$dependency]) && $this->relation[$dependency] === $dependent) {
+                // This relation has been already defined on other path. We mark
+                // it as circular reference.
+                NestedArray::setValue($this->tree, [
+                    ...$path,
+                    ...[$dependent],
+                ], $dependent . ':' . self::CIRCULAR_REFERENCE);
+                continue;
+            }
+
+            // Save this relation to avoid infinite circular references.
+            $this->relation[$dependency] = $dependent;
+
+            if (isset($this->dependents[$dependent])) {
+                $this->buildTree($dependent, $path);
+            } else {
+                NestedArray::setValue($this->tree, [...$path, ...[$dependent]], $dependent);
+            }
+        }
+    }
+
+    /**
+     * @param array $dependenciesPerDependent
+     * @return $this
+     */
+    protected function buildDependents(array $dependenciesPerDependent): self
+    {
+        foreach ($dependenciesPerDependent as $dependent => $dependencies) {
+            foreach ($dependencies as $dependency) {
+                $this->dependents[$dependency][$dependent] = $dependent;
+            }
+        }
+
+        // Make dependents order predictable.
+        foreach ($this->dependents as $dependency => $dependents) {
+            ksort($this->dependents[$dependency]);
+        }
+        ksort($this->dependents);
+
+        return $this;
+    }
+
+    /**
+     * @param bool $scanModuleDependencies
+     */
+    protected function scanConfigs(bool $scanModuleDependencies = true): void
+    {
+        $entityTypeManager = \Drupal::entityTypeManager();
+        $configTypeIds = array_keys(
+            array_filter($entityTypeManager->getDefinitions(), function (EntityTypeInterface $entityType): bool {
```

Comment:
could you improve readability with an arrow function here? https://medium.com/@albertcolom/how-to-use-arrow-function-in-php-c28490ff7fb7